Pagination/MongoTrait/FilterTrait.php

```diff
@@ -26,228 +25,38 @@ public function setFilterTypeManager(FilterTypeManager $filterTypeManager)
     }
 
     /**
-     * @param Stage               $qa
-     * @param FinderConfiguration $configuration
-     *
-     * @return Stage
-     */
-    protected function generateFilter(Stage $qa, FinderConfiguration $configuration)
-    {
-        if (null !== $configuration->getSearch()) {
-            $filterSearch = $this->generateSearchFilter($configuration);
-            if (null !== $filterSearch) {
-                $qa->match($filterSearch);
-            }
-        }
-
-        return $qa;
-    }
-
-    /**
-     * @param Stage               $qa
-     * @param FinderConfiguration $configuration
+     * @param PaginateFinderConfiguration $configuration
+     * @param Stage                       $qa
+     * @param array                       $searchTypes
      *
      * @return Stage
      */
-    protected function generatePreFilter(Stage $qa, FinderConfiguration $configuration)
+    protected function generateFieldFilter(PaginateFinderConfiguration $configuration, Stage $qa, array $searchTypes)
```

Comment:
@itkg-nanne : should introduce plural form in the function name as many filters are generated